pyop2/local_kernel.py

```diff
@@ -250,3 +246,25 @@ def _loopy_arguments(self):
         """Return the loopy arguments associated with the kernel."""
         return tuple(a for a in self.code.callables_table[self.name].subkernel.args
                      if isinstance(a, lp.ArrayArg))
+
+    @cached_property
+    def num_flops(self):
+        if self.flop_count is not None:
+            return self.flop_count
+        else:
+            if isinstance(self.code, lp.TranslationUnit):
+                prog = self.code.with_entrypoints(self.name)
+                knl = prog.default_entrypoint
+                warnings = list(knl.silenced_warnings)
+                warnings.extend(['insn_count_subgroups_upper_bound',
+                                 'get_x_map_guessing_subgroup_size',
+                                 'summing_if_branches_ops'])
+                knl = knl.copy(silenced_warnings=warnings,
+                               options=lp.Options(ignore_boostable_into=True))
+                knl = lp.fix_parameters(knl, layer=1)
+                prog = prog.with_kernel(knl)
+            else:
+                prog = self.code
```

Comment:
Can you explain what is going on here?

Reply:
I added my commit comments as actual comments now to make it more clear. I also think that actually self.code might always be a translation unit, I just wasn't sure because in the constructor we say that
`@validate_type(("code", (lp.LoopKernel, lp.TranslationUnit), TypeError))`. I changed this and will run all firedrake tests on this to make sure there is no special case.